streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -275,31 +259,70 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
                               final RecordConverter recordConverter) {
         for (final TopicPartition topicPartition : topicPartitions) {
             globalConsumer.assign(Collections.singletonList(topicPartition));
+            long offset;
             final Long checkpoint = checkpointFileCache.get(topicPartition);
             if (checkpoint != null) {
                 globalConsumer.seek(topicPartition, checkpoint);
+                offset = checkpoint;
             } else {
                 globalConsumer.seekToBeginning(Collections.singletonList(topicPartition));
+                final AtomicLong position = new AtomicLong();
```

Comment:
For the above call, was curious why we couldn't seek for all the topic partitions that are missing positions here, instead of doing one by one look-up?

Reply:
Well, I guess we want to restore partition by partition because the "restore listener" callback is easier to handle this way?